langfuse/model.py

```diff
@@ -73,17 +72,56 @@ def get_langchain_prompt(self):
         pass
 
     @staticmethod
-    def get_langchain_prompt_string(content: str):
+    def _get_langchain_prompt_string(content: str):
         return re.sub(r"\{\{(.*?)\}\}", r"{\g<1>}", content)
 
+    @staticmethod
+    def _compile_template_string(content: str, **kwargs) -> str:
```

Comment:
Did you try to implement that with regex? I would prefer that over looping ourselves through the string.